src/Gui/CommandFeat.cpp

```diff
@@ -136,33 +136,47 @@ StdCmdSendToPythonConsole::StdCmdSendToPythonConsole()
 
 bool StdCmdSendToPythonConsole::isActive(void)
 {
-    return (Gui::Selection().size() == 1);
+    //active only if either 1 object is selected or multiple subobjects from the same object
+    int size = Gui::Selection().size();
+    int xsize = size > 0 ? Gui::Selection().getSelectionEx()[0].getSubNames().size() : 0;
+    return (size == 1 || (size >= 1 && size == xsize));
 }
 
 void StdCmdSendToPythonConsole::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-
-    const std::vector<Gui::SelectionObject> &sels = Gui::Selection().getSelectionEx("*",App::DocumentObject::getClassTypeId(),true,true);
+    const std::vector<Gui::SelectionObject> &sels = Gui::Selection().getSelectionEx("*",App::DocumentObject::getClassTypeId(),true,false);
     if (sels.empty())
         return;
     const App::DocumentObject *obj = sels[0].getObject();
     QString docname = QString::fromLatin1(obj->getDocument()->getName());
     QString objname = QString::fromLatin1(obj->getNameInDocument());
     try {
-        QString cmd = QString::fromLatin1("obj = App.getDocument(\"%1\").getObject(\"%2\")").arg(docname,objname);
+        // clear variables from previous run, if any
+        QString cmd = QLatin1String("try:\n    del(doc,obj,shp,sub,subs)\nexcept:\n    pass\n");

```

Comment:
As far as I know the except statement in Python should not be empty. So, you have to use an exception type here:
`except Exception:`